spec/supabase_dart_v2.yml

```diff
@@ -0,0 +1,3024 @@
+        name: Parsing custom headers.
+        description: |
+          Any `headers` will be passed through to the function. A common pattern is to pass a logged-in user's JWT token as an Authorization header.
+        isSpotlight: true
+        code: |
+          ```dart
+          final res = await supabase.functions.invoke(
+            'hello',
+            body: {'foo': 'baa'},
+            headers: {
+              'Authorization': 'Bearer ${supabase.auth.currentSession?.accessToken}'
+            },
+          );
+          ```
+  - id: select
+    description: |
+      Perform a SELECT query on the table or view.
+    title: 'Fetch data: select()'
+    notes: |
+      - By default, Supabase projects will return a maximum of 1,000 rows. This setting can be changed in Project API Settings. It's recommended that you keep it low to limit the payload size of accidental or malicious requests. You can use `range()` queries to paginate through your data.
+      - `select()` can be combined with [Filters](/docs/reference/dart/using-filters)
+      - `select()` can be combined with [Modifiers](/docs/reference/dart/using-modifiers)
+      - `apikey` is a reserved keyword if you're using the [Supabase Platform](/docs/guides/platform) and [should be avoided as a column name](https://github.com/supabase/supabase/issues/5465).
+    examples:
+      - id: getting-your-data
+        name: Getting your data
+        isSpotlight: true
+        code: |
+          ```dart
+          final data = await supabase
+            .from('cities')
+            .select('name');
+          ```
+      - id: selecting-specific-columns
+        name: Selecting specific columns
+        description: You can select specific fields from your tables.
+        code: |
+          ```dart
+          final data = await supabase
+            .from('countries')
+            .select('''
+              name,
+              cities (
+                name
+              )
+            ''');
+          ```
+      - id: query-foreign-tables
+        name: Query foreign tables
+        description: If your database has relationships, you can query related tables too.
+        code: |
+          ```dart
+          final data = await supabase
+            .from('products')
+            .select('''
+              id,
+              supplier:supplier_id ( name ),
+              purchaser:purchaser_id ( name )
+            ''');
+          ```
+      - id: query-the-same-foreign-table-multiple-times
+        name: Query the same foreign table multiple times
+        description: |
+          Sometimes you will need to query the same foreign table twice.
+          In this case, you can use the name of the joined column to identify
+          which join you intend to use. For convenience, you can also give an
+          alias for each column. For example, if we had a shop of products,
+          and we wanted to get the supplier and the purchaser at the same time
```

Comment:
I think the example description doesn't match the code? example is about product but code is about messages

Reply:
Thanks for catching this. I have updated the code to the correct one.